host/schedule_test.go

```diff
@@ -0,0 +1,541 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package host
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	schedulepb "go.temporal.io/api/schedule/v1"
+	taskqueuepb "go.temporal.io/api/taskqueue/v1"
+	workflowpb "go.temporal.io/api/workflow/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	sdkclient "go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/converter"
+	"go.temporal.io/sdk/worker"
+	"go.temporal.io/sdk/workflow"
+
+	"go.temporal.io/server/common/log/tag"
+	"go.temporal.io/server/common/payload"
+	"go.temporal.io/server/common/persistence/sql/sqlplugin/sqlite"
+	esclient "go.temporal.io/server/common/persistence/visibility/store/elasticsearch/client"
+	"go.temporal.io/server/common/primitives/timestamp"
+	"go.temporal.io/server/common/searchattribute"
+	"go.temporal.io/server/service/worker/scheduler"
+)
+
+/*
+more tests to write:
+
+various validation errors
+
+overlap policies, esp. buffer
+
+last completion result and last error
+
+worker restart/long-poll activity failure:
+	get it in a state where it's waiting for a wf to exit, say with bufferone
+	restart the worker/force activity to fail
+	terminate the wf
+	check that new one starts immediately
+*/
+
+type (
+	scheduleIntegrationSuite struct {
+		*require.Assertions
+		IntegrationBase
+		hostPort      string
+		sdkClient     sdkclient.Client
+		worker        worker.Worker
+		esClient      esclient.IntegrationTestsClient
+		taskQueue     string
+		dataConverter converter.DataConverter
+	}
+)
+
+func TestScheduleIntegrationSuite(t *testing.T) {
+	flag.Parse()
+	suite.Run(t, new(scheduleIntegrationSuite))
+}
+
+func (s *scheduleIntegrationSuite) SetupSuite() {
+	s.setupSuite("testdata/schedule_integration_test_cluster.yaml")
+	s.hostPort = "127.0.0.1:7134"
```

Comment:
why not use default 7233?

Reply:
this is what all the integration tests use. the main logic is in `IntegrationBase.setupSuite` and `temporalImpl.FrontendGRPCAddress`. we should probably factor it out into something reusable. in another PR?

Reply:
👍